### ISSUE TITLE
Fabric 2.2 Chaincode Devmode Environment - Java chaincode

### ISSUE THREAD

**Issue by neosis** [samlinux/htsc#10]
Hi,

I went through the meet up video on youtube and found it very useful. It helped because official documentation was not present specially for version 2.2. So Thank you for that!

I tested entire flow with chaincode written and built in java. CORE_CHAINCODE_LOGLEVEL=debug CORE_PEER_TLS_ENABLED=false CORE_CHAINCODE_ID_NAME=test:0 java -jar ./test-cc-1.0-SNAPSHOT.jar -peer.address localhost:7052.

Post that I ran approve and commit command as documented and they were successful with no errors.

However when I ran invoke, I got following error -
Error: endorsement failure during invoke. response: status:500 message:"make sure the chaincode test has been successfully defined on channel xxx and try again: chaincode definition for 'test' exists, but chaincode is not installed"

Any ideas or pointers?

Best
Ashutosh

**Comment by samlinux** [comment]
Hi thank you for opening that issue.

Could you post the invoke command ?

**Comment by neosis** [comment]
Sure. I am using binary mode.
`peer chaincode invoke -o 127.0.0.1:7050 -C xxx -n test -c '{"Args":["init"]}' --isInit`
Error: endorsement failure during invoke. response: status:500 message:"make sure the chaincode test has been successfully defined on channel xxx and try again: chaincode definition for 'test' exists, but chaincode is not installed"

**Comment by samlinux** [comment]
It looks like that the chaincode is not running, I have not tried java for chaincode, I'm afraid I won't be able to help you here. But if you have solved the problem, please let me know and I will extend the example to that.

**Comment by neosis** [comment]
Sure. I am working on it and if I am able to solve it, I will post what I did here so that you can extend the example. Thank you for looking at it.

**Comment by neosis** [comment]
Hi

latest update from my side. My set up did not work when I used fabric version 2.2.0. I was not able to debug why. In fact it did not even work for go based chaincode. It gave same error. Some comments on internet suggested that we should install/instantiate the chaincode. But it would have defeated the purpose of having "dev mode". So rather than going that path, I tried with higher version of fabric 2.3.2 (because as your video pointed, the official doc has this section only for version 2.3 onwards). With version 2.3.2 the jave chaincode dev mode worked as expected. In fact in docker mode, I was able to stop and start java chaincode without bringing down rest of docker network. That gave added productivity as i could freely make changes to chaincode, bring it up and test it. With this I am closing this issue. Thanks for your help.
